packages/react-instantsearch-dom/src/core/createInstantSearchServer.js

```diff
@@ -22,40 +23,43 @@ const createSearchParametersCollector = accumulator => {
   };
 };
 
-const getSearchParameters = (indexName, searchParameters) => {
-  const sharedParameters = searchParameters
-    .filter(searchParameter => !hasMultipleIndices(searchParameter.context))
+const getSearchParameters = (indexName, widgets) => {
```

Comment:
The function used to only deal with `SearchParameters`.  We have introduced the computation of the metadata which doesn't make sense. We could create a dedicated function to derive the metadata from the collected values.